From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wyatt Neal <wyatt.neal+git@gmail.com>
Date: Mon, 15 Dec 2025 17:19:10 -0500
Subject: [PATCH] 8786: normalize username to lowercase on windows

on windows systems, you can get usernames in various versions of casing
which can cause problems because the username is stored case sensitively
in the project file. this change normalizes windows usernames to the
commonly accepted all lower case to try to mitigate most of these, but
it won't solve any existing projects that are already messed up. in
those cases, you can either do what's suggested in issue #43 if you have
a lot OR you can update the <project>.rep/project.prp where the OWNER
value=<whatever> you need as reported by the
System.getPropert('user.name') in Java.
---
 .../Utility/src/main/java/ghidra/util/SystemUtilities.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Ghidra/Framework/Utility/src/main/java/ghidra/util/SystemUtilities.java b/Ghidra/Framework/Utility/src/main/java/ghidra/util/SystemUtilities.java
index de27bb7cc2..f30c26f283 100644
--- a/Ghidra/Framework/Utility/src/main/java/ghidra/util/SystemUtilities.java
+++ b/Ghidra/Framework/Utility/src/main/java/ghidra/util/SystemUtilities.java
@@ -123,6 +123,13 @@ public class SystemUtilities {
 		if (slashIndex >= 0) {
 			uname = uname.substring(slashIndex + 1);
 		}
+
+		// on Windows, the username can come back in various cases if you're
+		// using AD or similar, this creates issues in the project file so we
+		// normalize to lowercase as it shouldn't impact anything else
+		if (System.getProperty("os.name").toLowerCase().contains("windows")) {
+			uname = uname.toLowerCase();
+		}
 		return uname;
 	}
 
-- 
2.45.1

